driver-core/src/main/java/com/datastax/driver/core/policies/ErrorAwarePolicy.java

```diff
@@ -0,0 +1,401 @@
+     */
+    @Override
+    public Iterator<Host> newQueryPlan(String loggedKeyspace, Statement statement) {
+        final Iterator<Host> childQueryPlan = childPolicy.newQueryPlan(loggedKeyspace, statement);
+
+        return new AbstractIterator<Host>() {
+            private Queue<Host> skipped;
+
+            @Override
+            protected Host computeNext() {
+                while (childQueryPlan.hasNext()) {
+                    Host host = childQueryPlan.next();
+
+                    if (errorTracker.isHostValid(host)) {
+                        return host;
+                    }
+
+                    if (skipped == null)
+                        skipped = new ArrayDeque<Host>();
+                    skipped.offer(host);
+                }
+
+                if (skipped != null && !skipped.isEmpty())
+                    return skipped.poll();
+
+                return endOfData();
+            }
+        };
+    }
+
+    @Override
+    public void onAdd(Host host) {
+        childPolicy.onAdd(host);
+    }
+
+    @Override
+    public void onUp(Host host) {
+        childPolicy.onUp(host);
+    }
+
+    @Override
+    public void onDown(Host host) {
+        childPolicy.onDown(host);
+    }
+
+    @Override
+    public void onRemove(Host host) {
+        childPolicy.onRemove(host);
+    }
+
+    /**
+     * Creates a new error aware policy builder given the child policy
+     * that the resulting policy should wrap.
+     *
+     * @param childPolicy the load balancing policy to wrap with error
+     *                    awareness.
+     * @return the created builder.
+     */
+    public static Builder builder(LoadBalancingPolicy childPolicy) {
+        return new Builder(childPolicy);
+    }
+
+    /**
+     * Closes the policy. Calls the child policy's {@link LoadBalancingPolicy#close()} method, and shuts down the updater
+     * service created for the update of the {@link ErrorAwarePolicyTracker}.
+     */
+    @Override
+    public void close() {
+        childPolicy.close();
+        updaterService.shutdown();
+    }
+
+    static class Builder {
+        private final LoadBalancingPolicy childPolicy;
+
+        private ErrorAwarePolicyTracker errorTracker;
+        private long updatePeriod = 1000;
+        private int errorsPerSecThreshold = 10;
+        private int retryPeriod = 5000;
+
+        public Builder(LoadBalancingPolicy childPolicy) {
+            this.childPolicy = childPolicy;
+        }
+
+        public Builder withUpdatePeriodMillis(long updatePeriodMillis) {
+            this.updatePeriod = updatePeriodMillis;
+            return this;
+        }
+
+        // Hosts that have a number of errors/sec higher than the threshold are ignored.
+        public Builder withInclusionThreshold(int errorsPerSecThreshold) {
+            this.errorsPerSecThreshold = errorsPerSecThreshold;
+            return this;
+        }
+
+        public Builder withErrorTracker(ErrorAwarePolicyTracker errorTracker) {
```

Comment:
Ah, I see, I did not notice that in the `ErrorAwarePolicy` constructor.  We should document that this is the default in the `Builder.withErrorTracker` method, as that is likely where users will look for documentation (probably already on your list of things to do as you mentioned you were working on docs).